webapp/components/AccountSettings.tsx

```diff
@@ -1,10 +1,202 @@
-import { useAppSelector } from '../hooks';
+import { getDownloadURL, ref, uploadBytes } from 'firebase/storage';
+import { useState } from 'react';
+import { ErrorDTO } from '../API/types/errors-types';
+import { isUserDTO, UpdateUserByIdDTO } from '../API/types/users-types';
+import { updateUser } from '../API/userHooks';
+import { useAppDispatch, useAppSelector } from '../hooks';
+import { populateUser } from '../slices/UserSlice';
 import { RootState } from '../store';
+import styles from '../styles/AccountSettings.module.scss';
+import { storage } from './Firebase/FirebaseInit';
 
 const AccountSettings = () => {
+	const dispatch = useAppDispatch();
+	const user = useAppSelector((state: RootState) => state.user?.value);
 	const userAuth = useAppSelector((state: RootState) => state.userAuth.value);
+	const [isSaving, setIsSaving] = useState<boolean>(false);
+	const [profileImage, setProfileImage] = useState<File | null>(null);
+	const [prevewImageUrl, setPrevewImageUrl] = useState<string | null>(null);
+	const [userName, setUserName] = useState<string>(user?.userName as string);
+	const [isSaveButtonDisabled, setIsSaveButtonDisabled] =
+		useState<boolean>(true);
+	const [selectedFile, setSelectedFile] = useState<File | null>(null);
 
-	return <div>{userAuth?.displayName}</div>;
+	const handleSubmit = () => {
+		console.log('handle this');
+	};
+
+	const handleImageChange = (file: File) => {
+		if (!file) return;
+
+		setProfileImage(file);
+		setIsSaveButtonDisabled(false);
+
+		const reader = new FileReader();
+		reader.onloadend = () => {
+			setPrevewImageUrl(reader.result as string);
+		};
+		reader.readAsDataURL(file!);
+	};
+
+	const handleSave = async () => {
+		if (isSaveButtonDisabled) return;
+
+		setIsSaving(true);
+
+		if (profileImage) {
+			// if profile picture is updated, update it along with name in firebase and database
+			try {
+				const imageFirebaseRef = `/userProfilePics/${userAuth?.email}`;
+				const profileImageStorageRef = ref(storage, imageFirebaseRef);
+				const snapshot = await uploadBytes(
+					profileImageStorageRef,
+					profileImage
+				);
+				const downloadURL = await getDownloadURL(snapshot.ref);
+
+				if (!downloadURL) {
+					alert('Failed to save image to firebase');
+				} else {
+					const updateUserRes: UpdateUserByIdDTO | ErrorDTO = await updateUser({
+						userId: user?.userId as number,
+						userName: userName,
+						profileImgUrl: downloadURL,
+					});
+
+					if (!isUserDTO(updateUserRes)) {
```

Comment:
this line should be !isUpdateUserByIdDTO(updateUserRes). The isUpdateUserByIdDTO has not been implemented in the types folder. Look at type guard comment above.